Highway.cpp

```diff
@@ -13,7 +13,22 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    // assert(false);
+
+    if (Car* car = dynamic_cast<Car*>(v))
+    {
+        car->closeWindows();
+    }
+
+    if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v))
+    {
+        motorcycle->lanesplitAndRace();
+    }
+
+    if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v))
+    {
+        semiTruck->toSpeedLimit(s);
+    }
```

Comment:
use `else-if`
you're still checking if it's a Motorcycle, even if it's a Car
- [ ] revise